libopae/plugins/xfpga/sysfs.c

```diff
@@ -73,31 +73,56 @@ static uint32_t _sysfs_region_count;
 
 #define SYSFS_FORMAT(s) sysfs_path_table[_sysfs_format_index].s
 
-typedef enum { SYSFS_FME = 0, SYSFS_PORT } resource_type;
 
-typedef struct sysfs_region sysfs_fpga_region;
+#define SYSFS_MAX_REGIONS 128
+static sysfs_fpga_region _regions[SYSFS_MAX_REGIONS];
 
-typedef struct fpga_resource {
-	sysfs_fpga_region *region;
-	char path[SYSFS_PATH_MAX];
-	resource_type type;
-	int num;
-} sysfs_fpga_resource;
+#define PCIE_PATH_PATTERN "([0-9a-fA-F]{4}):([0-9a-fA-F]{2}):([0-9]{2})\\.([0-9])/fpga"
+#define PCIE_PATH_PATTERN_GROUPS 5
 
-#define SYSFS_MAX_RESOURCES 4
-typedef struct sysfs_region {
-	char path[SYSFS_PATH_MAX];
-	int number;
-	sysfs_fpga_resource *fme;
-	sysfs_fpga_resource *port;
-} sysfs_fpga_region;
+#define PARSE_REGEX_INT(_p, _m, _v, _b) \
```

Comment:
This isn't actually doing a regex parse.  Should it be renamed to clarify?

Reply:
Ok 👍